packages/eslint-config-react/index.js

```diff
@@ -13,8 +13,6 @@ module.exports = {
     "react/no-access-state-in-setstate": "error",
     "react/prefer-es6-class": ["error", "always"],
     "react/no-deprecated": "error",
-    "react/no-did-mount-set-state": "error",
```

Comment:
Its ok to call setState, but should we maybe mark this as a warning? Most probably there is a better way of doing this. @Schedulaar what do you think?

Reply:
I think a warning is okay because you have to be careful with it, but I don't think there is always a better solution.

Reply:
Let's go for a warning :+1: